File: visitor/java/Encyclopedia.java
package Visitor;

// UML diyagramındaki Element'ten türer.
// Accept metoduna bulunduğu nesne örneği gönderilir.
// UML diyagramındaki ConcreteElement yapısına denk gelir.
class Encyclopedia extends Book {
    @Override
    public void accept(IBookExportVisitor visitor) {
        visitor.exportVisit(this);
    }
}

File: visitor/python/encyclopedia.py
from book import Book


class Encyclopedia(Book):
    """
    UML diyagramındaki Element'ten türer.
    Accept metoduna bulunduğu nesne örneği gönderilir.
    UML diyagramındaki ConcreteElement yapısına denk gelir.
    """

    def accept(self, visitor):
        visitor.export_visit(self)
